pinot-common/src/main/java/org/apache/pinot/common/request/BrokerRequest.java

```diff
@@ -1975,71 +1956,71 @@ public void read(org.apache.thrift.protocol.TProtocol iprot, BrokerRequest struc
             }
             break;
           case 13: // DEBUG_OPTIONS
             if (schemeField.type == org.apache.thrift.protocol.TType.MAP) {
               {
-                org.apache.thrift.protocol.TMap _map97 = iprot.readMapBegin();
-                struct.debugOptions = new java.util.HashMap<java.lang.String,java.lang.String>(2*_map97.size);
-                @org.apache.thrift.annotation.Nullable java.lang.String _key98;
-                @org.apache.thrift.annotation.Nullable java.lang.String _val99;
-                for (int _i100 = 0; _i100 < _map97.size; ++_i100)
+                org.apache.thrift.protocol.TMap _map105 = iprot.readMapBegin();
+                struct.debugOptions = new java.util.HashMap<java.lang.String,java.lang.String>(2*_map105.size);
+                @org.apache.thrift.annotation.Nullable java.lang.String _key106;
+                @org.apache.thrift.annotation.Nullable java.lang.String _val107;
+                for (int _i108 = 0; _i108 < _map105.size; ++_i108)
                 {
-                  _key98 = iprot.readString();
-                  _val99 = iprot.readString();
-                  struct.debugOptions.put(_key98, _val99);
+                  _key106 = iprot.readString();
+                  _val107 = iprot.readString();
+                  struct.debugOptions.put(_key106, _val107);
                 }
                 iprot.readMapEnd();
               }
               struct.setDebugOptionsIsSet(true);
-            } else {
+            } else { 
               org.apache.thrift.protocol.TProtocolUtil.skip(iprot, schemeField.type);
             }
             break;
           case 14: // QUERY_OPTIONS
             if (schemeField.type == org.apache.thrift.protocol.TType.MAP) {
               {
-                org.apache.thrift.protocol.TMap _map101 = iprot.readMapBegin();
-                struct.queryOptions = new java.util.HashMap<java.lang.String,java.lang.String>(2*_map101.size);
-                @org.apache.thrift.annotation.Nullable java.lang.String _key102;
-                @org.apache.thrift.annotation.Nullable java.lang.String _val103;
-                for (int _i104 = 0; _i104 < _map101.size; ++_i104)
+                org.apache.thrift.protocol.TMap _map109 = iprot.readMapBegin();
+                struct.queryOptions = new java.util.HashMap<java.lang.String,java.lang.String>(2*_map109.size);
+                @org.apache.thrift.annotation.Nullable java.lang.String _key110;
+                @org.apache.thrift.annotation.Nullable java.lang.String _val111;
+                for (int _i112 = 0; _i112 < _map109.size; ++_i112)
                 {
-                  _key102 = iprot.readString();
-                  _val103 = iprot.readString();
-                  struct.queryOptions.put(_key102, _val103);
+                  _key110 = iprot.readString();
+                  _val111 = iprot.readString();
+                  struct.queryOptions.put(_key110, _val111);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 2009:2021, 1989:2001.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.